### PR TITLE
Nathansen/create homework

### PR DESCRIPTION
**Pull request by AndreaNathansen**
Added homework section including detailed view

### CODE REVIEW

**Comment by kremer-io** [comment]
@AndreaNathansen Due the PR from @JonasWanke, we've some conflicts.

File: app/src/main/res/layout/item_homework.xml [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+                <TextView
+                    android:layout_width="match_parent"
+                    android:layout_height="wrap_content"
+                    android:layout_marginTop="8dp"
+                    android:ellipsize="end"
+                    android:gravity="start"
+                    android:maxLines="4"
+                    android:textSize="@dimen/text_medium_size"
+                    android:text="@{HomeworkViewHolder.getTextFromHtml(homework.description)}"
```
Comment (JonasWanke) [inline app/src/main/res/layout/item_homework.xml]: For showing HTML, please use either ContentWebView or PassiveWebView (the latter in this case). TextView doesn't support all HTML features. Also the HTML-parsing blocks the UI thread which can cause dropped frames. WebView takes care of this!

File: app/src/main/java/org/schulcloud/mobile/controllers/main/HomeworkListAdapter.kt [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+                    }
+                    1 -> {
+                        return "Morgen"
+                    }
+                    Int.MAX_VALUE -> {
+                        return ""
+                    }
+                    else -> {
+                        return try {
+                            DateTimeFormat.forPattern("dd.MM.yyyy").print(homework.getDueTillDateTime())
```
Comment (JonasWanke) [inline app/src/main/java/org/schulcloud/mobile/controllers/main/HomeworkListAdapter.kt]: DateUtils.formatDateTime() from native Android can format dates too and also takes into account the user's language

File: app/src/main/java/org/schulcloud/mobile/controllers/main/HomeworkListAdapter.kt [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+                        } catch (e: IllegalArgumentException) {
+                            ""
```
Comment (JonasWanke) [inline app/src/main/java/org/schulcloud/mobile/controllers/main/HomeworkListAdapter.kt]: Don't ignore exceptions!

File: app/src/main/java/org/schulcloud/mobile/controllers/main/HomeworkListFragment.kt [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        activity?.title = getString(R.string.homework_title)
+        return inflater.inflate(R.layout.fragment_homework_list, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        swipeRefreshLayout = swipeRefresh
+
+        viewModel.homework.observe(this, Observer {
+            homeworkListAdapter.update(it ?: emptyList())
```
Comment (JonasWanke) [inline app/src/main/java/org/schulcloud/mobile/controllers/main/HomeworkListFragment.kt]: This list is sorted ascending with regard to the due date. While this makes sense for yet-to-be-completed homework, old ones still appear at the top (e.g. login with the demo-teacher, and there's a homework from 2017 at the top).

File: app/src/main/res/layout/activity_homework.xml [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+            android:id="@+id/swipeRefresh"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content">
+
+            <ScrollView
+                android:layout_width="match_parent"
+                android:layout_height="match_parent"
+                android:scrollbars="none">
+
+                <LinearLayout
```
Comment (JonasWanke) [inline app/src/main/res/layout/activity_homework.xml]: Same as above regarding nested layouts

File: app/src/main/res/layout/activity_homework.xml [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+
+                    </LinearLayout>
+
+                    <TextView
+                        style="@style/TextAppearance.AppCompat.Medium"
+                        android:layout_width="match_parent"
+                        android:layout_height="wrap_content"
+                        android:layout_marginTop="5dp"
+                        android:gravity="start"
+                        android:text="@{viewModel.homework.title}"
```
Comment (JonasWanke) [inline app/src/main/res/layout/activity_homework.xml]: The title should be shown inside the toolbar for consistency.

File: app/src/main/java/org/schulcloud/mobile/controllers/main/HomeworkListFragment.kt (hunk 4 above)
Comment (AndreaNathansen) [reply app/src/main/java/org/schulcloud/mobile/controllers/main/HomeworkListFragment.kt]: About the required behaviour: shall the old ones be shown at the end or not shown at all?

File: app/src/main/res/layout/item_homework.xml (hunk 1 above)
Comment (AndreaNathansen) [reply app/src/main/res/layout/item_homework.xml]: For the homework preview I need to limit the number of lines displayed, as well as adding an ellipsize at the end. WebView doesn't support this.

File: app/src/main/java/org/schulcloud/mobile/controllers/main/HomeworkListAdapter.kt (hunk 2 above)
Comment (AndreaNathansen) [reply app/src/main/java/org/schulcloud/mobile/controllers/main/HomeworkListAdapter.kt]: I changed it to DateTimeFormat.mediumDate(), that function also formats depending on the locale. DateUtils.formatDateTime() wouldn't work in a static function as it requires a context as an argument.

File: app/src/main/java/org/schulcloud/mobile/controllers/main/HomeworkListFragment.kt (hunk 4 above)
Comment (AndreaNathansen) [reply app/src/main/java/org/schulcloud/mobile/controllers/main/HomeworkListFragment.kt]: This is fixed now. I don't know why GitHub still doesn't display it as outdated

File: app/src/main/res/layout/activity_homework.xml (hunk 6 above)
Comment (AndreaNathansen) [reply app/src/main/res/layout/activity_homework.xml]: Fixed

Comment (JonasWanke) [reply app/src/main/res/layout/activity_homework.xml]: But now you show the title in both the toolbar and the content itself. Could you please remove it here? (Should be the last change to be made)

Comment (AndreaNathansen) [reply app/src/main/res/layout/activity_homework.xml]: Shall I remove it from the toolbar or from the content?

Comment (JonasWanke) [reply app/src/main/res/layout/activity_homework.xml]: From the content

Comment (AndreaNathansen) [reply app/src/main/res/layout/activity_homework.xml]: Fixed. But now that isn't consistent with the IOS version anymore

**Pull request merged**